src/inference/hmc.jl

```diff
@@ -262,6 +466,41 @@ function step(model, spl::Sampler{<:Hamiltonian}, vi::VarInfo, is_first::Val{fal
     return vi, is_accept
 end
 
+
+# Function for multiple steps
+function steps!(model, spl::Sampler{<:AdaptiveHamiltonian}, vi, samples; rng::AbstractRNG=GLOBAL_RNG)
+    ahmc_samples =  AHMC.sample(rng, spl.info[:h], spl.info[:traj], Vector{Float64}(vi[spl]), spl.alg.n_iters, spl.info[:adaptor], spl.alg.n_adapts)
+    for i = 1:length(samples)
+        vi[spl] = ahmc_samples[i]
+        samples[i].value = Sample(vi, spl).value
+    end
+end
+
+function steps!(model, spl::Sampler{<:HMC}, vi, samples; rng::AbstractRNG=GLOBAL_RNG)
+    ahmc_samples =  AHMC.sample(rng, spl.info[:h], spl.info[:traj], Vector{Float64}(vi[spl]), spl.alg.n_iters)
+    for i = 1:length(samples)
+        vi[spl] = ahmc_samples[i]
+        samples[i].value = Sample(vi, spl).value
```

Comment:
`vi.logp` will be the same for all?

Reply:
When `Sample(vi, spl)` is called, it automatically copies the `logp` from `vi` to `value[:lp]`.

Reply:
But unless the model is called on `vi`, this `logp` will be 0? Shouldn't the `logp` values be obtained from AHMC side?

Reply:
Oh sorry I see your point. I thought this comment was on the non-AHMC one. The interface there doesn't return `logp` so far. We may want to add it later.